lichess-org/lila
# Refactor tournament confetti

- convert confetti to a typescript implementation
- uses [canvas-confetti](https://github.com/catdad/canvas-confetti)
- `disableForReducedMotion: true` to disable confetti entirely for users that prefer reduced motion

### To test locally:

in `ui/tournament/src/view/finished.ts`, add "`true ||`" to the conditional to always see the confetti:

```diff
function confetti(data: TournamentData): VNode | undefined {
-    if (data.me && data.isRecentlyFinished && once('tournament.end.canvas.' + data.id))
+    if (true || data.me && data.isRecentlyFinished && once('tournament.end.canvas.' + data.id))
    return h('canvas#confetti', {
```

then visit a finished arena.

## Before + After

1st half of video is before, 2nd half is after:


https://github.com/user-attachments/assets/ee1d5f23-922e-4c3b-bef9-963781805cd1
